### PR TITLE
add ANKISYNCD_COMMIT

### PR DESCRIPTION
**Pull request by senorsmile**
This adds the ability to checkout a specific commit of anki-sync-server as well as specify the version of the image to be built.

Idea partially taken from this now outdated PR: https://github.com/ankicommunity/anki-devops-services/pull/12/files.

### CODE REVIEW

File: services/anki-sync-server/images/Dockerfile [hunk 1]
```diff
@@ -7,10 +7,13 @@ WORKDIR ${ANKISYNCD_ROOT}
 
 COPY bin/download-release.sh ./bin/download-release.sh
 
+ARG ANKISYNCD_COMMIT='latest'
+ENV ANKISYNCD_COMMIT=${ANKISYNCD_COMMIT}
```
Comment (VikashKothary) [inline services/anki-sync-server/images/Dockerfile:11]: Is there a benefit here of defining this as an environment variable? As it's never used by the software or won't be changed by the user.

Maybe using `LABEL` might be better here? 
This PR had a similar idea so see my comment there: https://github.com/ankicommunity/anki-devops-services/pull/12/files#r563159827

**Review by VikashKothary — COMMENTED** [review]
Thanks so much for this, This is a great idea.

Please review my comment and let me know what you think.

File: services/anki-sync-server/images/Dockerfile (hunk 1 above)
Comment (senorsmile) [reply services/anki-sync-server/images/Dockerfile:11]: For my use case, I will always override this in CI/CD, so I need to be able specify a separate version.  That's the primary reason for adding the ANKISYNCD_COMMIT value.

Comment (senorsmile) [reply services/anki-sync-server/images/Dockerfile:11]: Actually, I may have mixed up how ARG and ENV work (again).  If we remove the ENV line, we won't be able to see it from the live running container (which we don't care about) but can still override it at image build time?

Comment (VikashKothary) [reply services/anki-sync-server/images/Dockerfile:11]: Hi, that is correct. The ARG value will still be available at build time therefore will still be available when running `download-release.sh`. And therefore removing the ENV line won't affect that functionality of being able to specifc  the version in your CI build.

That being said, if you build two different images with two different commits, there's no way to check this in the final image. As such, I'd recommend using `LABEL` here so you can then check it using `docker inspect`

Comment (VikashKothary) [reply services/anki-sync-server/images/Dockerfile]: I've replaced the ENV with a LABEL to unblock this PR. If you think this is a mistake, we can create a new issue to discuss this further.

Thank you for this contribution.

**Review by VikashKothary — APPROVED** [review]

**Pull request merged**